src/main/drivers/compass/compass_spi_hmc5883l.c

```diff
@@ -21,17 +21,22 @@
 
 #include <platform.h>
 
+#include "config/parameter_group.h"
+
 #include "drivers/io.h"
 #include "drivers/bus_spi.h"
 
 #include "compass.h"
 #include "compass_hmc5883l.h"
 
+#define SPI_CFG_TO_DEV(x) ((x) - 1)
+
 #ifdef USE_MAG_SPI_HMC5883
 
 #define DISABLE_HMC5883      IOHi(hmc5883CsPin)
 #define ENABLE_HMC5883       IOLo(hmc5883CsPin)
 
+static SPI_TypeDef *hmc5883SpiInstance;
 static IO_t hmc5883CsPin = IO_NONE;
 
 bool hmc5883SpiWriteCommand(uint8_t reg, uint8_t data)
```

Comment:
If we are moving away from using a hard-coded `HMC5883_SPI_INSTANCE` then I think `hmc5883SpiInstance` should be a parameter to `hmc5883SpiWriteCommand` so we have:
`bool hmc5883SpiWriteCommand(SPI_TypeDef *hmc5883SpiInstance, uint8_t reg, uint8_t data)`.

Once we have done that, it is no longer HMC5883 specific, so we have:
`bool spiWriteCommand(SPI_TypeDef *spiInstance, uint8_t reg, uint8_t data)` which can then be moved to `bus_spi.c`.

Furthermore we should not have a `static` `hmc5883SpiInstance`, rather `spiInstance` should be part of `busDevice_t` (along with the current `csnPin`).

And finally we should not use a static `hmc5883CsPin`, rather we should use the `csnPin` in `busDevice_t`.